### PR TITLE
go-e API: neuer Befehl zur Stromvorgabe

### PR DESCRIPTION
**Pull request by mutsch75**
Siehe https://openwb.de/forum/viewtopic.php?f=5&t=1789

Mqtt Befehl abhängig von FW Version

### CODE REVIEW

**Review by benderl — CHANGES_REQUESTED** [review]
Kannst Du die Einrückung bitte korrigieren?

**Comment by benderl** [comment]
Wie sieht die Ausgabe des curl Aufrufs für den Status aus? Die brauche ich einmal, um die Abfrage durchspielen zu können.

**Comment by mutsch75** [comment]
pi@OpenWB:~ $ curl --silent --connect-timeout 2 -s http:/192.168.178.117/mqtt?payload=amx=10                                                                                                                                                 {"version":"B","tme":"1404211010","rbc":"14","rbt":"1530411415","car":"2","amp":"10","err":"0","ast":"0","alw":"1","stp":"0","cbl":"20","pha":"63","tmp":"2","tma":[12.13,11.00,11.13,12.75],"amt":"32","dws":"684373","dwo":"0","adi":"1","uby":"0","eto":"4900","wst":"3","txi":"0","nrg":[221,221,218,2,100,97,0,22,22,0,0,447,100,100,3,64],"fwv":"040.0","sse":"022616","wss":"TheAir123","wke":"**********","wen":"1","cdi":"0","tof":"101","tds":"1","lbr":"255","aho":"3","afi":"7","azo":"0","ama":"32","al1":"16","al2":"18","al3":"20","al4":"24","al5":"32","cid":"255","cch":"16711935","cfi":"65280","lse":"0","ust":"1","wak":"b95d5623ad","r1x":"0","dto":"0","nmo":"0","sch":"AAAAAAAAAAAAAAAA","sdp":"0","eca":"0","ecr":"0","ecd":"0","ec4":"0","ec5":"0","ec6":"0","ec7":"0","ec8":"0","ec9":"0","ec1":"0","rca":"048E1C7A","rcr":"","rcd":"","rc4":"","rc5":"","rc6":"","rc7":"","rc8":"","rc9":"","rc1":"","rna":"","rnm":"","rne":"","rn4":"","rn5":"","rn6":"","rn7":"","rn8":"","rn9":"","rn1":"","loe":0,"lot":17,"lom":6,"lop":1,"log":"PFCGZWSC","lon":0,"lof":0,"loa":0,"lch":0,"mce":0,"mcs":"test.mosquitto.org","mcp":1883,"mcu":"","mck":"","mcc":0}

**Comment by mutsch75** [comment]
Sorry du wolltest ja den Status

**Comment by mutsch75** [comment]
pi@OpenWB:~ $ curl --silent --connect-timeout 2 -s http:/192.168.178.117/status
{"version":"B","tme":"1404211015","rbc":"14","rbt":"1530708437","car":"2","amp":"11","err":"0","ast":"0","alw":"1","stp":"0","cbl":"20","pha":"63","tmp":"2","tma":[12.50,11.50,11.38,13.63],"amt":"32","dws":"818835","dwo":"0","adi":"1","uby":"0","eto":"4900","wst":"3","txi":"0","nrg":[219,222,219,2,109,107,0,24,24,0,0,488,100,100,3,66],"fwv":"040.0","sse":"022616","wss":"TheAir","wke":"**********","wen":"1","cdi":"0","tof":"101","tds":"1","lbr":"255","aho":"3","afi":"7","azo":"0","ama":"32","al1":"16","al2":"18","al3":"20","al4":"24","al5":"32","cid":"255","cch":"16711935","cfi":"65280","lse":"0","ust":"1","wak":"b95d5623ad","r1x":"0","dto":"0","nmo":"0","sch":"AAAAAAAAAAAAAAAA","sdp":"0","eca":"0","ecr":"0","ecd":"0","ec4":"0","ec5":"0","ec6":"0","ec7":"0","ec8":"0","ec9":"0","ec1":"0","rca":"048E1C7A","rcr":"","rcd":"","rc4":"","rc5":"","rc6":"","rc7":"","rc8":"","rc9":"","rc1":"","rna":"","rnm":"","rne":"","rn4":"","rn5":"","rn6":"","rn7":"","rn8":"","rn9":"","rn1":"","loe":0,"lot":17,"lom":6,"lop":1,"log":"PFCGZWSC","lon":0,"lof":0,"loa":0,"lch":0,"mce":0,"mcs":"test.mosquitto.org","mcp":1883,"mcu":"","mck":"","mcc":0}                                                pi@OpenWB:~ $

File: goecheck.sh [hunk 1]
```diff
@@ -17,10 +17,16 @@ goecheck(){
 					curl --silent --connect-timeout $goetimeoutlp1 -s http://$goeiplp1/mqtt?payload=alw=0 > /dev/null
 				fi
 			fi
+			fwv=$(echo $output | jq -r '.fwv')
+			printf -v fwv %0.f "$fwv"
```
Comment (benderl) [inline goecheck.sh:21]: Wurde das in einer aktuellen Installation geprüft? Ich erhalte immer die Fehlermeldung `printf: 040.0: Ungültige Oktalzahl`.

Comment (mutsch75) [reply goecheck.sh:21]: Ja, kann den output leider nicht eliminieren, das Ergebniss stimmt aber.

Comment (mutsch75) [reply goecheck.sh:21]: Alternativ fwv=$(echo $fmw | grep -Po "[1-9]\d{1,2}")

Comment (mutsch75) [reply goecheck.sh:21]: Vielleicht ist so tatsächlich die bessere Variante

fwv=$(echo $output | jq -r '.fwv' | grep -Po "[1-9]\d{1,2}")

**Comment by benderl** [comment]
Die Anpassungen sind noch nicht vollständig.
https://openwb.de/forum/viewtopic.php?p=31681#p31681

**Review by benderl — APPROVED** [review]

**Comment by benderl** [comment]
Kannst Du die Änderungen bitte auch in der `runs/set-current.sh` ab Zeile 192 einfügen? Die Zusammenhänge der Dateien sind etwas undurchsichtig. Die Stelle habe ich zumindest noch mit dem alten Aufruf gefunden.

**Comment by mutsch75** [comment]
Erledigt

**Pull request merged**